images/win/scripts/Installers/Validate-ToolCache.ps1

```diff
@@ -17,85 +17,114 @@ function Get-ToolcachePackages {
-                        }
-                        else
-                        {
-                            Write-Host "$env:AGENT_TOOLSDIRECTORY\$SoftwareName\$foundVersion does not include required architecture"
-                            exit 1
-                        }
-                    }
-                    else
-                    {
-                        Write-Host "$env:AGENT_TOOLSDIRECTORY\$SoftwareName\$version.* was not found"
-                        exit 1
-                    }
-                }
-
-                Add-SoftwareDetailsToMarkdown -SoftwareName $SoftwareName -DescriptionMarkdown $description
-            }
+        if (Test-Path "$Path\$test")
+        {
+            Write-Host "$SoftwareName($test) $SoftwareVersion($SoftwareArchitecture) is successfully installed:"
+            Write-Host (& "$Path\$test" --version)
         }
         else
         {
-            Write-Host "$env:AGENT_TOOLSDIRECTORY\$SoftwareName does not include any folders"
+            Write-Host "$SoftwareName($test) $SoftwareVersion($SoftwareArchitecture) is not installed"
             exit 1
         }
     }
-    else
+}
+
+function UpdateMarkdownDescription {
+    param (
+        [string]$Description,
+        [Parameter(Mandatory = $True)]
+        [string]$SoftwareVersion,
+        [Parameter(Mandatory = $True)]
+        [string]$SoftwareArchitecture
+    )
+    return $Description += "_Version:_ $SoftwareVersion ($SoftwareArchitecture)<br/>"
+}
+
+function ToolcacheTest {
+    param (
+        [Parameter(Mandatory = $True)]
+        [string]$SoftwareName,
+        [Parameter(Mandatory = $True)]
+        [string[]]$ExecTests
+    )
+
+    $markdownDescription = ""
+    $softwarePath = "$env:AGENT_TOOLSDIRECTORY\$SoftwareName"
+
+    if (-Not (Test-Path $softwarePath))
+    {
+        Write-Host "$softwarePath does not exist"
+        exit 1
+    }
+
+    [array]$installedVersions = GetChildFolders -Path $softwarePath
+    if ($installedVersions.count -eq 0)
     {
-        Write-Host "$env:AGENT_TOOLSDIRECTORY\$SoftwareName does not exist"
+        Write-Host "$softwarePath does not include any folders"
         exit 1
     }
+
+    $tools = GetToolsByName -SoftwareName $SoftwareName
+    foreach($tool in $tools)
+    {
+        foreach ($version in $tool.Versions)
+        {
+            $foundVersion = $installedVersions | where { $_.StartsWith($version) }
+            if ($foundVersion -eq $null)
+            {
+                Write-Host "$softwarePath\$version.* was not found"
+                exit 1
+            }
+
+            $installedArchitecture = GetChildFolders -Path "$softwarePath\$foundVersion"
+            $requiredArchitecture = $tool.Architecture
+            if (-Not ($installedArchitecture -Contains $requiredArchitecture))
+            {
+                Write-Host "$softwarePath\$foundVersion does not include required architecture"
```

Comment:
let's add to this error message what architecture we were looking for and failed to find.

Reply:
Thanks for the point!